shared/rakefiles/secrets.rb

```diff
@@ -119,12 +123,28 @@ def self.push_secrets(secrets, encryption_key)
     gs_bucket = "#{ENV['TF_VAR_project_id']}-#{encryption_key}-secrets"
     encoded_secrets = Base64.encode64(secrets.to_json).delete!("\n")
 
+    puts "[secret-mgmt] Retrieving primary key version for key '#{encryption_key}'..."
+    encryption_key_version = %x{
+      curl -s \
+      -H \"Authorization:Bearer $(gcloud auth print-access-token)\" \
+      -H \"Content-Type:application/json\" \
+      -X GET \"#{Secrets::GOOGLE_KMS_API}/v1/projects/#{ENV['TF_VAR_project_id']}/locations/#{Secrets::KMS_LOCATION}/keyRings/#{Secrets::KMS_KEYRING}/cryptoKeys/#{encryption_key}\"
+    }
+
+    begin
+      encryption_key_version = JSON.parse(encryption_key_version)
+      encryption_key_version = encryption_key_version['primary']['name'].match(/\/([0-9]+)$/)[1]
```

Comment:
I find this hard to understand. Some things that might help me:
* An example of what `primary.name` looks like in the returned JSON blob (`"name": "projects/gpii-gcp-dev-tyler/locations/global/keyRings/keyring/cryptoKeys/default/cryptoKeyVersions/1"`).
* Moving the `['name'].match(/\/([0-9]+)$/)[1]` part (which is repeated three times in this file) to a function `get_latest_version_from_key_name`.

(In general I prefer changing the delimiter over escaping, but here I'm undecided on whether I think it's harder to read `%r{}` or a single `\/`.)

Reply:
Good suggestion!